update msgpack to v4.1.3 (#644)
cpp-pm/hunter: docs/packages/pkg/msgpack.rst

-  Added by `Antal Tátrai <https://github.com/tatraian>`__
   (`pr-406 <https://github.com/ruslo/hunter/pull/406>`__)
- Available since |hunter|

> - Target library name is now changed from "msgpack::msgpack" to "msgpackc-cxx"

Reviewer: Please reference the package version number since when the target name was changed (in this case it is `v4.1.3`)

Reviewer: ```suggestion
- Target library name from "msgpack::msgpack" to "msgpackc-cxx" with `v4.1.3`
```

    hunter_add_package(msgpack)
    find_package(msgpack CONFIG REQUIRED)

>     target_link_libraries(... msgpackc-cxx)

Reviewer: please add a note about the changed target name in the documentation
